src/plugins/aws/ssm/index.ts

```diff
@@ -246,12 +262,40 @@ const spawnSsmNode = async (
         return;
       }
 
-      options.abortController.abort(code);
+      options.abortController?.abort(code);
       print2(`SSH session terminated`);
       resolve(code);
     });
   });
 
+/**
+ * Spawns a child process to add a private key to the ssh-agent. The SSH agent is included in the OpenSSH suite of tools
+ * and is used to hold private keys during a session. The SSH agent typically does not persist keys across system reboots
+ * or logout/login cycles. Once you log out or restart your system, any keys added to the SSH agent during that session
+ * will need to be added again in subsequent sessions.
+ */
+const executeScpCommand = async (
+  credential: AwsCredentials,
+  command: string,
+  privateKey: string
+) => {
+  // Execute should not leave any ssh-agent processes running after it's done performing an SCP transaction.
+  const execute = `
```

Comment:
Things started to get really complex when I tried to run all these commands as separate processes. I would have needed to find a way to share environment variables across sub-processes, make sure that tty was working for scp, and then do the cleanup after SCP was done (more abort controller shenanigans). Instead it's easier to just execute the commands in the order we need them.